documents/write_acls.go

```diff
@@ -59,6 +63,7 @@ func getChangedFields(oldTree, newTree *proofs.DocumentTree, lengthSuffix string
 
 		if !bytes.Equal(ov, nv) {
 			changedFields = append(changedFields, changedField{
+				name:     k,
```

Comment:
please modify the first for loop above to be
```
props := make(map[string]proofs.Property)
	for _, p := range append(oldProps, newProps...) {
		// we can ignore the length property since any change in slice or map will return in addition or deletion of properties in the new tree
		if p.Text == lengthSuffix {
			continue
		}

		if _, ok := props[p.ReadableName()]; ok {
			continue
		}

		props[p.ReadableName()] = p

		_, ol := oldTree.GetLeafByProperty(p.ReadableName())
		_, nl := newTree.GetLeafByProperty(p.ReadableName())

		if ol == nil {
			changedFields = append(changedFields, newChangedField(p, nl, false))
			continue
		}

		if nl == nil {
			changedFields = append(changedFields, newChangedField(p, ol, true))
			continue
		}

		ov := ol.Value
		nv := nl.Value
		if ol.Hashed {
			ov = ol.Hash
			nv = nl.Hash
		}

		if !bytes.Equal(ov, nv) {
			changedFields = append(changedFields, changedField{
				property: p.CompactName(),
				old:      ov,
				new:      nv,
			})
		}
	}
```
and remove second.